### ISSUE TITLE
Fine-tuning DANNs

### ISSUE THREAD

**Issue by Nikhil0504** [galactic-ai/haloflow#18]

**Comment by Nikhil0504** [comment]
Started fine-tuning using Weights and Biases. See, https://wandb.ai/nikhil0504/dann_sweep/sweeps/voux5aby?nw=nwusernikhil0504 for the hyperparameter sweep (needs permission).

**Comment by Nikhil0504** [comment]
The old link doesn't work. I broke the runs into three sweeps: [mags](https://wandb.ai/nikhil0504/dann_sweep/sweeps/30pj7e3w?nw=nwusernikhil0504), [mags_morph](https://wandb.ai/nikhil0504/dann_sweep/sweeps/r2rc99n8) and [mags_morph_extra](https://wandb.ai/nikhil0504/dann_sweep/sweeps/84fskxcl)

@changhoonhahn you can access the link now. It also has the TNSE plots for each of the hyperparameter runs

**Comment by changhoonhahn** [comment]
@Nikhil0504 Very interesting. FYI the mag_morph link doesn't work for me.

Why do some of these TSNE plots have bizarre shapes? 

The domain adaption works very poorly for mag_morph_extra. Do you understand this?

**Comment by Nikhil0504** [comment]
> Why do some of these TSNE plots have bizarre shapes?

It's likely because of the early stopping.  But, also TNSE plots are _confusing_ to interpret. See, https://wandb.ai/nikhil0504/dann_sweep/runs/kqajg4gq?nw=nwusernikhil0504 for an example. The test loss decreased exponentially and the last epoch had a value of 2. The domain accuracy remained consistent at about 0.342. 

> The domain adaption works very poorly for mag_morph_extra. Do you understand this?

I don't I would say it works very poorly because the scatter plot below shows a wide distribution of ranges from 0.46 to 0.32.
<img width="491" alt="Image" src="https://github.com/user-attachments/assets/2112517d-5a26-4333-a8be-759bef2f5513" />

The `mags` sweep had a range of about 0.46 to 0.35. That's the other extreme case because we don't provide any morphology. 

I grouped the test simulations for all the runs/sweeps and plotted the domain accuracy. No matter what observable we use, the domain accuracy highly depends on the test simulation we keep.

<img width="1136" alt="Image" src="https://github.com/user-attachments/assets/97e5f9c6-aa91-475e-a250-771865c5bfe0" />

**Comment by changhoonhahn** [comment]
Hmmm, let's dig deeper into this. We want to make sure we understand what's going on well. 

I suspect one issue is that we're optimizing on cross-entropy but our assessment is based on domain accuracy. For the training runs that didn't immediately fail, check the cross-entropy losses. Let's see if they're being properly optimized. 

Also another sanity check: (1) randomly assign labels, (2) calculate cross-entropy and domain accuracy (3) compare to cross-entropy and domain accuracy in your runs.

**Comment by Nikhil0504** [comment]
Fixed the issue with the losses. See, 4039942ea8190e4ae8c25a396df8e8149893b9a0

Main change was implementing `alpha` correctly.
https://github.com/galactic-ai/haloflow/blob/4039942ea8190e4ae8c25a396df8e8149893b9a0/bin/dann/train.py#L85-L86

Also, we can't use `TNG_ALL` simulation since it uses both `TNG50` and `TNG100`.
Here is a TNSE plot with 
```python
# Configuration
sims = ['TNG50', 'TNG100', 'Eagle100', 'Simba100']
config = {
    'obs': 'mags_morph',
    'train_sim': sims[:-1],
    'batch_size': 32,
    'num_epochs': 100,
    'lr': 0.001,
    'feature_layers': [128, 64, 32], 
    'label_layers': [32, 16, 8], 
    'domain_layers': [32, 16, 8], 
    'es_patience': 5,
    'es_min_delta': 0.001
}

```
![Image](https://github.com/user-attachments/assets/722e312a-db27-4a7a-a3df-39a30e3c88fd)
